tasks/files/snapshot.py

```diff
@@ -202,18 +202,83 @@ def lvm_full_report_json():
     return lvm_json
 
 
-def get_snapshot_name(lv_name, prefix, suffix):
-    if prefix:
-        prefix_str = prefix
-    else:
-        prefix_str = ""
+def lvm_get_fs_mount_points(block_path):
+    find_mnt_command = [
+        "findmnt",
+        block_path,
+        "-P",
+    ]
+
+    rc, output = run_command(find_mnt_command)
+    if rc:
+        return None
+
+    output = output.replace('"', "")
+    output = output.replace("\n", "")
+    kv = dict(arg.split("=", 1) for arg in output.split(" ") if arg)
```

Comment:
I think this function could just do
```
return dict(arg.split("=", 1) for arg in output.split(" ") if arg)
```
That is, return a `dict` instead of a string.  I think the function that calls `lvm_get_fs_mount_points` will ultimately do the json string conversion to return to the caller

Reply:
>  it isn't working properly for blockdevices that are mounted in multiple mount points

Ah - then I guess you could do the `output.replace('"', "")`, then instead of `output.replace("\n", "")` do `lines = output.split("\n")`, then iterate over `lines` and return a `list` of `dict` from this function

Reply:
I've updated so the output for the case of multiple mounts looks like:

```
    "mounts": {
        "/dev/vg3/lv1_vg3": null,
        "/dev/vg3/lv1_vg3_snapset2": [
            {
                "TARGET": "/mnt/database",
                "SOURCE": "/dev/mapper/vg3-lv1_vg3_snapset2",
                "FSTYPE": "xfs",
                "OPTIONS": "rw,relatime,seclabel,attr2,inode64,logbufs=8,logbsize=32k,noquota"
            }
        ],
        "/dev/vg3/lv2_vg3": null,
        "/dev/vg3/lv2_vg3_snapset2": null,
        "/dev/vg3/lv3_vg3": null,
        "/dev/vg3/lv3_vg3_snapset2": null,
        "/dev/vg2/lv1_vg2": null,
        "/dev/vg2/lv1_vg2_snapset2": [
            {
                "TARGET": "/mnt/production_mnt",
                "SOURCE": "/dev/mapper/vg2-lv1_vg2_snapset2",
                "FSTYPE": "xfs",
                "OPTIONS": "rw,relatime,seclabel,attr2,inode64,logbufs=8,logbsize=32k,noquota"
            }
        ],
        "/dev/vg2/lv2_vg2": null,
        "/dev/vg2/lv2_vg2_snapset2": null,
        "/dev/vg1/lv1_vg1": null,
        "/dev/vg1/lv1_vg1_snapset2": [
            {
                "TARGET": "/mnt/new_mountpoint",
                "SOURCE": "/dev/mapper/vg1-lv1_vg1_snapset2",
                "FSTYPE": "xfs",
                "OPTIONS": "rw,relatime,seclabel,attr2,inode64,logbufs=8,logbsize=32k,noquota"
            },
            {
                "TARGET": "/mnt/other_mp",
                "SOURCE": "/dev/mapper/vg1-lv1_vg1_snapset2",
                "FSTYPE": "xfs",
                "OPTIONS": "rw,relatime,seclabel,attr2,inode64,logbufs=8,logbsize=32k,noquota"
            }
        ],
        "/dev/vg1/lv2_vg1": null,
        "/dev/vg1/lv2_vg1_snapset2": null
    }
```

I think I followed you suggestion and the result it:
```

def lvm_get_fs_mount_points(block_path):
    find_mnt_command = [
        "findmnt",
        block_path,
        "-P",
    ]
    mount_list = list()

    rc, output = run_command(find_mnt_command)
    if rc:
        return None

    output = output.replace('"', "")

    for line in output.split("\n"):
        if len(line):
            mount_list.append(dict(arg.split("=", 1) for arg in line.split(" ") if arg))

    return mount_list
```